### PR TITLE
ENG-171: Journal replay integrity check + PR review fixes

### PR DESCRIPTION
**Pull request by Connorbelez**
## Summary

Implements journal replay integrity verification for the cash ledger and fixes 24 issues found during PR review across ENG-161 through ENG-171.

### Implementation (ENG-171)
- Read-only integrity checker that replays journal entries in canonical `sequenceNumber` order, derives expected account balances, and compares against stored `cumulativeDebits`/`cumulativeCredits`
- Detects balance drift and missing sequence gaps
- Public query (`journalReplayIntegrityCheck`) for admin on-demand checks
- Internal query for cron integration
- Cursor management for incremental replay tracking
- Layer 2 check in daily reconciliation action (07:00 UTC cron)
- 23 tests covering drift, gaps, scopes, reversals, and edge cases

### PR Review Fixes (this PR)

**CRITICAL** (4):
- `publish` RBAC corrected — was `docGenMutation`, changed to `adminMutation` (template publishing is a design operation)
- `dailyReconciliation` now has try-catch around each layer — failures no longer silently kill the cron
- `lookupStatus` now handles all 14 entity types (was 4) with error logging for any future gaps
- `postCashReceiptForObligation` missing receivable now logs error + audit entry (was `console.warn` + null)

**IMPORTANT** (8):
- Phase 1 data volume comment on unbounded `.collect()` in replay integrity
- Incremental mode clarified as accumulation-only (no verification performed)
- Missing account null-checks added in `getPostingGroupSummary` and `getControlBalancesByPostingGroup`
- Cursor regression guard in `advanceReplayCursor`
- Layer 1/2 renamed to StatusCheck/BalanceCheck to avoid CLAUDE.md terminology conflict

**SUGGESTIONS** (12):
- `SerializedReplayMismatch` type exported for DRY between modules
- Runtime guard in `filterByScope` throws when both scope filters set
- `ServicingFeeMetadata` index signature replaced with `extends Record` for `Record<string, unknown>` assignability
- `isReplayPassing` pure predicate exported
- JSDoc on `accumulateEntries` and `compareAgainstStored`
- Stale ENG-156 TODO removed (SUSPENSE routing already implemented)
- `cashReceipt.test.ts` harness updated to register `auditLog` component

## Test plan
- [x] `bun run test` — 67/67 cashLedger tests pass
- [x] `bun typecheck` — no errors in changed files
- [x] `bun check` — no errors in changed files

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add journal replay integrity verification to the cash ledger and integrate it into daily reconciliation while tightening error handling, access control, and auditability across related payment and document flows.

New Features:
- Introduce a read-only journal replay integrity checker for the cash ledger with public and internal entry points for scoped or full verification.
- Expose an admin-accessible query to trigger journal replay integrity checks on demand.

Bug Fixes:
- Ensure daily reconciliation layers run independently with error handling so a failure in one layer does not stop the other.
- Expand reconciliation status lookups to cover all supported entity types and log unhandled types instead of silently skipping them.
- Update cash receipt posting for obligations to log errors and emit audit entries when the receivable account is missing instead of silently skipping work.
- Prevent control-balance and posting-group calculations from using journal entries that reference missing accounts by logging and skipping those entries.
- Tighten RBAC for document template publishing so only admins can publish templates.

Enhancements:
- Track and advance a replay cursor for incremental journal integrity runs with guards against cursor regression.
- Log journal replay integrity outcomes and surface mismatches and missing sequences as reconciliation discrepancies mapped to synthetic cash-ledger entities.
- Refine servicing fee metadata typing to be assignable to generic record types for better type safety and extensibility.
- Add stricter runtime guards for replay scope filtering to avoid ambiguous combinations of scope parameters.
- Export reusable replay integrity helper types and predicates for use across modules.

Tests:
- Add an extensive replayIntegrity test suite covering clean runs, drift detection, sequence gaps, scoping, reversals, cursor behavior, and helper utilities.
- Update cash receipt tests to register the auditLog component so new audit behaviors are exercised.

## Summary by CodeRabbit

* **New Features**
  * Added cash-ledger replay integrity checks (full & incremental) with cursor advancement and a public integrity check endpoint.

* **Bug Fixes**
  * Stronger error handling, logging, and audit entries when ledger accounts are missing; receipts route to suspense when needed.
  * Posting-group validation now flags corrupt/skipped entries.
  * Daily reconciliation upgraded to two guarded layers for improved discrepancy capture.
  * Publish action now requires admin-level authorization.

* **Tests**
  * Extensive integration and unit tests covering replay integrity, gaps, scope filtering, idempotency, and edge cases.

### CODE REVIEW

File: convex/payments/cashLedger/replayIntegrity.ts [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+/**
+ * Serialized variant of ReplayMismatch for use across Convex query/action boundaries
+ * where the typed Id<> cannot be serialized directly. The `accountId` is widened to string.
+ */
+export type SerializedReplayMismatch = Omit<ReplayMismatch, "accountId"> & {
+	accountId: string;
+};
+
+export interface ReplayResult {
+	accountsChecked: number;
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/cashLedger/replayIntegrity.ts]: Updated the docstring to accurately describe behavior: a runtime guard throws when both accountId and mortgageId are set, rather than incorrectly stating accountId takes precedence. The throw is the correct safety-first behavior.

File: convex/payments/cashLedger/replayIntegrity.ts [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+// ── Scope Filtering ──────────────────────────────────────────
+
+/**
+ * Filter journal entries by scope. Returns only entries that match the
+ * given accountId or mortgageId constraints. If neither is specified,
+ * all entries pass through.
+ *
+ * NOTE: When both accountId and mortgageId are provided, accountId takes
+ * precedence and mortgageId is ignored. A runtime guard throws if both are set.
+ */
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/cashLedger/replayIntegrity.ts]: Same fix as Thread 1 (same docstring issue on line 84-85): updated the docstring to correctly state that a runtime guard throws when both accountId and mortgageId are set, rather than incorrectly claiming accountId takes precedence. The throw behavior is intentional and correct.
